### PR TITLE
BGDIINF_SB-2530: add a first draft of readiness probe

### PR DESCRIPTION
**Pull request by ltclm**
* readiness probe, status will be written to checker_ready.txt in shared volume or
tmp folder

TODO:
- [x] update infra-vhost https://github.com/geoadmin/infra-vhost/pull/118
- [x] add HTTP Readiness Probe to service-search-wsgi https://github.com/geoadmin/service-search-wsgi/pull/43

### CODE REVIEW

File: scripts/checker.sh [hunk 1]
```diff
@@ -3,5 +3,51 @@ set -e
...
+        rm -f "${READY_FILE}" || :
+        rm -f "${BOOTSTRAP_FILE}" || :
+}
+
+# source this stuff until here
+[ "$0" = "${BASH_SOURCE[*]}" ] || return 0
+
+# step 1
+# skip probes during the bootstrap of the container, initial rsync
+[ -e "${BOOTSTRAP_FILE}" ] || exit 0
```
Comment (ltshb) [inline scripts/checker.sh:34]: Do I understand this correctly, we exit with 0 when `${BOOTSTRAP_FILE}` does not exist ?

File: scripts/docker-cmd.sh [hunk 2]
```diff
@@ -38,7 +44,19 @@ rsync --update --delete -av --exclude "*.tmp.*" --stats ${SPHINXINDEX_EFS} ${SPH
...
 echo -e "${green}starting searchd service ...${NC}"
-/usr/bin/searchd --nodetach "$@"
\ No newline at end of file
+/usr/bin/searchd  "$@"
+
+# this inifite loop is necessary to trap docker signals SIGHUP SIGTERM
+# and do the clean-up in the volume with the trap function
+while true; do
+    sleep 10
+done
```
Comment (ltshb) [inline scripts/docker-cmd.sh:62]: Note sure I understand it correctly here. Line 56 is starting searchd in background ? When docker sends the SIGUP or SIGTERM signal ?

File: scripts/docker-cmd.sh [hunk 3]
```diff
@@ -38,7 +44,19 @@ rsync --update --delete -av --exclude "*.tmp.*" --stats ${SPHINXINDEX_EFS} ${SPH
 service cron start || exit 1
 crontab < docker-crontab
 
+# once the initial sync and index creation is done the docker volume is ready / bootstrapped
+# from now on the liveness probe is checking the searchd status
+touch "${BOOTSTRAP_FILE}"
+echo "bootstrap finished $(date +"%F %T"), start precaching indexes" > "${READY_FILE}"
+
```
Comment (ltshb) [inline scripts/docker-cmd.sh:51]: I would not create the ready file here, because we are not ready.

**Review by ltshb — COMMENTED** [review]
Some

**Comment by ltclm** [comment]
@ltshb as discussed on the phone. i have simplified the probes here.

File: scripts/checker.sh [hunk 4]
```diff
@@ -3,5 +3,20 @@ set -e
...
-# Do the lifeness check exit status 0 -> success | 1 -> fail
-searchd --status 1> /dev/null && exit 0 || exit 1
\ No newline at end of file
+# mountpoint with readiness probe file
+MOUNT=$(realpath "${PROBE_MOUNTPOINT:-/tmp}")
+READY_FILE="${MOUNT}/checker_ready.txt"
+
+clean_probe_files() {
+        rm -f "${READY_FILE}" || :
+}
```
Comment (ltshb) [inline scripts/checker.sh:12]: indent

File: scripts/checker.sh [hunk 5]
```diff
@@ -3,5 +3,20 @@ set -e
...
+
+# source this stuff until here
+[ "$0" = "${BASH_SOURCE[*]}" ] || return 0
+
+if searchd --status 1> /dev/null; then
+    echo "READY" > "${READY_FILE}"
+    exit 0
+else
+    exit 1
```
Comment (ltshb) [inline scripts/checker.sh:21]: Here we should remove the ready file to mark the service as un ready when unhealthy

Comment (ltclm) [reply scripts/checker.sh:21]: good point, yes!

**Comment by ltclm** [comment]
@ltshb need a review again, thank you!

**Review by ltshb — APPROVED** [review]
💯

**Pull request merged**